apps/server/src/modules/h5p-editor/contentStorage/h5p-content.entity.ts

```diff
@@ -0,0 +1,119 @@
+import { Embeddable, Embedded, Entity, JsonType, Property } from '@mikro-orm/core';
+import { IContentMetadata, ILibraryName } from '@lumieducation/h5p-server';
+import { BaseEntity } from '@shared/domain';
+import { IContentAuthor, IContentChange } from '@lumieducation/h5p-server/build/src/types';
+
+@Embeddable()
+export class ContentMetadata implements IContentMetadata {
+	@Property({ nullable: true })
+	dynamicDependencies?: ILibraryName[];
+
+	@Property({ nullable: true })
+	editorDependencies?: ILibraryName[];
+
+	@Property()
+	embedTypes: ('iframe' | 'div')[];
+
+	@Property({ nullable: true })
+	h?: string;
+
+	@Property()
+	language: string;
+
+	@Property()
+	mainLibrary: string;
+
+	@Property({ nullable: true })
+	metaDescription?: string;
+
+	@Property({ nullable: true })
+	metaKeywords?: string;
+
+	@Property()
+	preloadedDependencies: ILibraryName[];
+
+	@Property({ nullable: true })
+	w?: string;
+
+	@Property()
+	defaultLanguage: string;
+
+	@Property({ nullable: true })
+	a11yTitle?: string;
+
+	@Property()
+	license: string;
+
+	@Property({ nullable: true })
+	licenseVersion?: string;
+
+	@Property({ nullable: true })
+	yearFrom?: string;
+
+	@Property({ nullable: true })
+	yearTo?: string;
+
+	@Property({ nullable: true })
+	source?: string;
+
+	@Property()
+	title: string;
+
+	@Property({ nullable: true })
+	authors?: IContentAuthor[];
+
+	@Property({ nullable: true })
+	licenseExtras?: string;
+
+	@Property({ nullable: true })
+	changes?: IContentChange[];
+
+	@Property({ nullable: true })
+	authorComments?: string;
+
+	@Property({ nullable: true })
+	contentType?: string;
+
+	constructor(metadata: IContentMetadata) {
+		this.embedTypes = metadata.embedTypes;
+		this.language = metadata.language;
+		this.mainLibrary = metadata.mainLibrary;
+		this.defaultLanguage = metadata.defaultLanguage;
+		this.license = metadata.license;
+		this.title = metadata.title;
+		this.preloadedDependencies = metadata.preloadedDependencies;
+
+		if (metadata.dynamicDependencies) this.dynamicDependencies = metadata.dynamicDependencies;
```

Comment:
I'm wondering why you added these if-conditions... If the respective prop is undefined, they don't make a difference, do they? Because afterwards the prop of the entity will be undefined either way. They would make a difference for empty strings, but is that something that needs to be handled?